packages/contract-schema/test/solc.js

```diff
@@ -89,7 +90,7 @@ contract B {
 
     //check that ast and legacyAST have the correct form
     assert.equal(A.ast.nodeType, "SourceUnit");
-    assert.equal(A.legacyAST.name, "SourceUnit");
+    assert.equal(A.legacyAST.nodeType, "SourceUnit");
```

Comment:
this stands out as being suspect to me... can we verify that the `legacyAST` is actually the legacy AST format?

Reply:
Hey @gnidan . Thanks for the comment. I tried printing out the `legacyAST` which gives me the below format in JSON.
`{
  absolutePath: 'A.sol',
  exportedSymbols: { A: [ 12 ], B: [ 17 ] },
  id: 18,
  license: 'MIT',
  nodeType: 'SourceUnit',
  nodes: [
    {
      id: 1,
      literals: [Array],
      nodeType: 'PragmaDirective',
      src: '34:23:0'
    },
    {
      abstract: false,
      baseContracts: [],
      canonicalName: 'A',
      contractDependencies: [],
      contractKind: 'contract',
      fullyImplemented: true,
      id: 12,
      linearizedBaseContracts: [Array],
      name: 'A',
      nameLocation: '70:1:0',
      nodeType: 'ContractDefinition',
      nodes: [Array],
      scope: 18,
      src: '61:80:0',
      usedErrors: []
    },
    {
      abstract: false,
      baseContracts: [],
      canonicalName: 'B',
      contractDependencies: [],
      contractKind: 'contract',
      fullyImplemented: true,
      id: 17,
      linearizedBaseContracts: [Array],
      name: 'B',
      nameLocation: '154:1:0',
      nodeType: 'ContractDefinition',
      nodes: [Array],
      scope: 18,
      src: '145:60:0',
      usedErrors: []
    }
  ],
  src: '34:174:0'
}`
If this is what you are mentioning about or please do correct me if I am wrong somewhere.

Reply:
hm that looks like the new AST format. could you compare and see?

if newer Solidity gets rid of the legacy format, we probably want to remove it from our tests, and possibly see about removing that field from @truffle/contract in such cases (otherwise we'd be lying)

Reply:
I think they have removed the legacy AST stuff from `0.8.0` onwards. I will check that. Thanks.

Reply:
we can probably treat it as a separate PR to fix the contract behavior, but we probably want to remove this line from the test here (if it is, in fact, removed.)

thanks @sukanyaparashar!

cc @haltman-at in case you have opinions when you get back

Reply:
Is this helpful? https://docs.soliditylang.org/en/v0.8.13/080-breaking-changes.html?highlight=legacyAST#interface-changes

Reply:
Yes @gnidan . I think I can remove this line in this test and then make another PR for removing the legacy AST in the `@truffle/contract` as both `A.ast` and `A.legacyAST` are yielding the same results. We can wait for @haltman-at 's comment on this.

Reply:
> Is this helpful? https://docs.soliditylang.org/en/v0.8.13/080-breaking-changes.html?highlight=legacyAST#interface-changes

Thanks @cds-amal . This link is helpful. Is the `compact AST` mentioned there same as we are referring to in `A.ast`?

Reply:
> Whether that's a good idea, I don't know, but just confirming that that is the current behavior, even if it ought to be changed.

Definitely should be changed! That's a bug :)

Reply:
Sure, @gnidan and @haltman-at . We can change this behavior in another PR.